service/src/test/java/bio/terra/workspace/service/workspace/AwsCloudContextConnectedTest.java

```diff
@@ -0,0 +1,179 @@
+
+    // Iterate over supported regions in Environment.
+    for (Region region : environment.getSupportedRegions()) {
+      Optional<LandingZone> landingZoneOptional = environment.getLandingZone(region);
+      assertTrue(landingZoneOptional.isPresent());
+
+      LandingZone landingZone = landingZoneOptional.get();
+      assertNotNull(landingZone.getMetadata(), region + ": landingZone.metadata null");
+      assertNotNull(landingZone.getStorageBucket(), region + ": landingZone.storageBucket null");
+      assertNotNull(landingZone.getKmsKey(), region + ": landingZone.kmsKey null");
+      assertFalse(
+          landingZone.getNotebookLifecycleConfigurations().isEmpty(),
+          region + ": landingZone.notebookLifecycleConfigurations empty");
+
+      // Log details about the landing zone.
+      logMetadata(landingZone.getMetadata(), "Landing zone region: " + region);
+
+      // Perform a HeadBucket command to check storage bucket existence; this requires ListBucket
+      // permissions on the bucket, which the WSM role has.  This is a basic "Hello, world!" test
+      // that we've discovered the environment and assumed the TerraWorkspaceManager role.
+
+      String bucketName = landingZone.getStorageBucket().name();
+      HeadBucketRequest request = HeadBucketRequest.builder().bucket(bucketName).build();
+
+      // SdkClient classes should generally be used in try-with-resource blocks.
+      try (S3Client s3Client =
+          S3Client.builder().region(region).credentialsProvider(awsCredentialsProvider).build()) {
+        s3Client.headBucket(request);
+      } catch (Exception exception) {
+        fail(String.format("HeadBucket request failed: %s", exception), exception);
+      }
+
+      logger.info("Confirmed access to bucket {} in AWS Region {}", bucketName, region);
+    }
+  }
+
+  @Test
+  void createCloudContextTest() {
+    AwsCloudContext createdCloudContext =
+        awsCloudContextService.createCloudContext(
+            "flightId", WorkspaceFixtures.DEFAULT_SPEND_PROFILE_ID);
+    assertNotNull(createdCloudContext);
+
+    Metadata envMetadata = awsCloudContextService.discoverEnvironment().getMetadata();
+    AwsCloudContextFields contextFields = createdCloudContext.getContextFields();
+    assertNotNull(contextFields);
+    assertEquals(envMetadata.getMajorVersion(), contextFields.getMajorVersion());
+    assertEquals(envMetadata.getOrganizationId(), contextFields.getOrganizationId());
+    assertEquals(envMetadata.getAccountId(), contextFields.getAccountId());
+    assertEquals(envMetadata.getTenantAlias(), contextFields.getTenantAlias());
+    assertEquals(envMetadata.getEnvironmentAlias(), contextFields.getEnvironmentAlias());
+
+    CloudContextCommonFields commonFields = createdCloudContext.getCommonFields();
+    assertNotNull(commonFields);
+    assertEquals("flightId", commonFields.flightId());
+    assertThat(commonFields.spendProfileId(), equalTo(WorkspaceFixtures.DEFAULT_SPEND_PROFILE_ID));
```

Comment:
Out of curiosity, when do we use assertEquals vs assertThat... equalTo?

Reply:
+1, please use assertEqual

Reply:
I didnt see a pattern / guidance on this in the rest of tests. The one I generally followed - 
assertEquals used when comparing individual fields / temp objects
assertThat used when comparing result / etc to a static object. 

No concerns sticking to assertEquals, changed